packages/xstate-react/src/useSelector.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { ActorRef, Interpreter, Subscribable } from 'xstate';
+import { isActorWithState } from './useActor';
+import { getServiceSnapshot } from './useService';
+
+export function useSelector<
+  TActor extends ActorRef<any, any>,
+  T,
+  TEmitted = TActor extends Subscribable<infer Emitted> ? Emitted : never
+>(
+  actor: TActor,
+  selector: (emitted: TEmitted) => T,
+  compare: (a: T, b: T) => boolean = (a, b) => a === b,
+  getSnapshot: (a: TActor) => TEmitted = (a) =>
+    a instanceof Interpreter
```

Comment:
I would prefer if we could not rely on the `instanceof` - it might create very hard to debug problems for users. That's why we've created a special property on the StateNode in the past that has allowed us to differentiate those objects. 

A recent Twitter thread mentioning this problem: https://twitter.com/_jayphelps/status/1356992397243256837